rust/protocol/src/sealed_sender.rs

```diff
@@ -1074,31 +1440,30 @@ pub async fn sealed_sender_decrypt_to_usmc(
             encrypted_message,
         } => {
             let eph_keys = sealed_sender_v1::EphemeralKeys::calculate(
+                &our_identity.into(),
                 &ephemeral_public,
-                our_identity.public_key(),
-                our_identity.private_key(),
-                false,
+                Direction::Receiving,
             )?;
 
             let message_key_bytes = crypto::aes256_ctr_hmacsha256_decrypt(
                 &encrypted_static,
-                eph_keys.cipher_key()?,
-                eph_keys.mac_key()?,
+                eph_keys.cipher_key.as_ref(),
+                eph_keys.mac_key.as_ref(),
             )?;
 
             let static_key = PublicKey::try_from(&message_key_bytes[..])?;
 
             let static_keys = sealed_sender_v1::StaticKeys::calculate(
+                &our_identity,
                 &static_key,
-                our_identity.private_key(),
-                eph_keys.chain_key()?,
+                &eph_keys.chain_key,
                 &encrypted_static,
             )?;
 
             let message_bytes = crypto::aes256_ctr_hmacsha256_decrypt(
                 &encrypted_message,
-                static_keys.cipher_key()?,
-                static_keys.mac_key()?,
+                static_keys.cipher_key.as_ref(),
+                static_keys.mac_key.as_ref(),
```

Comment:
`&`

Reply:
Done! Got to this before seeing this comment too!